src/DFG.cpp

```diff
@@ -133,28 +191,123 @@ void DFG::combineCmpBranch() {
   }
 }
 
-void DFG::combineMulAdd() {
+// Combine phi + iadd or phi + iadd + iadd where iadd is integer addition.
+void DFG::combinePhiAdd(int type) {
+  // detect patterns (e.g., mul+alu)
+  DFGNode* phiNode = NULL;
+  DFGNode* addNode = NULL;
+  DFGNode* addNode2 = NULL;
+  bool found = false;
+  for (DFGNode* dfgNode: nodes) {
+    if (dfgNode->isPhi() and !dfgNode->hasCombined()) {
+      found = false;
+      for (DFGNode* succNode: *(dfgNode->getSuccNodes())) {
+        if (found) break;
+        if (succNode->isIadd() and !succNode->hasCombined()) {
+          for (DFGNode* succNode2: *(succNode->getSuccNodes())) {
+            if (succNode2->isIadd() and !succNode2->hasCombined()) {
+              phiNode = dfgNode;
+              phiNode->setCombine(type);
+              addNode = succNode;
+              phiNode->addPatternPartner(addNode);
+              addNode->setCombine(type);
+              addNode2 = succNode2;
+              phiNode->addPatternPartner(addNode2);
+              addNode2->setCombine(type);
+              found = true;
+              break;
+            }
+          }
+        }
+      }
+    }
+  }
+  for (DFGNode* dfgNode: nodes) {
```

Comment:
Plz add comment for this second for loop, why is it needed and how is it cooperate with the first one.